packages/web3-core/src/web3_request_manager.ts

```diff
@@ -288,7 +289,20 @@ export class Web3RequestManager<
 					const responsePromise = result as unknown as Promise<
 						JsonRpcResponse<ResponseType>
 					>;
-					responsePromise.then(resolveWithResponse).catch(rejectWithError);
+					responsePromise.then(resolveWithResponse).catch(error => {
+							try {
+								// Attempt to process the error response
+								const processedError = this._processJsonRpcResponse(
+									payload,
+									error as JsonRpcResponse<ResponseType, unknown>,
+									{ legacy: true, error: true }
+								);
+								reject(processedError);
```

Comment:
According to codevov, this line is not checked at unit testing.

Reply:
covered